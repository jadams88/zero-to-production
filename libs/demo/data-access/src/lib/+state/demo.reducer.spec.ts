import { demoReducer } from './demo.reducer';
import * as ExampleActions from './demo.actions';
import { IExample } from '../example.interface';

describe('ExampleReducer', () => {
  describe('undefined action', () => {
    it('should return the default state', () => {
      const action = {} as any;
      const result = demoReducer(undefined, action);
      expect(result).toMatchSnapshot();
    });
  });

  describe('AddExamples', () => {
    it('should add the demo to the example state', () => {
      const examples: IExample[] = [
        {
          title: 'some title',
          summary: 'some description',
          description: 'some full description',
          url: 'example-1',
        },
        {
          title: 'another title',
          summary: 'some other description',
          description: 'some other full description',
          url: 'example-2',
        },
      ];

      const action = ExampleActions.addExamples({ examples });
      const result = demoReducer(undefined, action);
      expect(result).toMatchSnapshot();
    });
  });
});
