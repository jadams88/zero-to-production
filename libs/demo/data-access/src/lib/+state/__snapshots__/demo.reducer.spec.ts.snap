// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExampleReducer AddExamples should add the demo to the example state 1`] = `
Object {
  "examples": Object {
    "entities": Object {
      "example-1": Object {
        "description": "some full description",
        "summary": "some description",
        "title": "some title",
        "url": "example-1",
      },
      "example-2": Object {
        "description": "some other full description",
        "summary": "some other description",
        "title": "another title",
        "url": "example-2",
      },
    },
    "ids": Array [
      "example-1",
      "example-2",
    ],
    "selectedExampleUrl": null,
  },
}
`;

exports[`ExampleReducer undefined action should return the default state 1`] = `
Object {
  "examples": Object {
    "entities": Object {},
    "ids": Array [],
    "selectedExampleUrl": null,
  },
}
`;
